oak_attestation_verification/src/verifier.rs

```diff
@@ -761,9 +763,31 @@ fn verify_hex_digests(actual: &HexDigest, expected: &HexDigest) -> anyhow::Resul
     }
 }
 
-fn verify_text(actual: &str, expected: &TextReferenceValue) -> anyhow::Result<()> {
+fn verify_text(
+    now_utc_millis: i64,
+    actual: &str,
+    expected: &TextReferenceValue,
+    endorsement: Option<&TransparentReleaseEndorsement>,
+) -> anyhow::Result<()> {
     match expected.r#type.as_ref() {
         Some(text_reference_value::Type::Skip(_)) => Ok(()),
+        Some(text_reference_value::Type::Endorsement(public_keys)) => {
+            let endorsement =
+                endorsement.context("matching endorsement not found for text reference value")?;
+            verify_binary_endorsement(
+                now_utc_millis,
+                &endorsement.endorsement,
+                &endorsement.endorsement_signature,
+                &endorsement.rekor_log_entry,
+                &public_keys.endorser_public_key,
+                &public_keys.rekor_public_key,
+            )?;
+            // Compare the actual command line against the one inlined in the endorsement.
+            if actual.as_bytes() != &endorsement.subject {
```

Comment:
I thought the endorsement attachment would contain a regex as well, so I would expect to check a regex match here.

Reply:
Also, shouldn't this be the attachement rather than the subject?

Reply:
Agree. How does this work with TB's size issue, if we compare via RegExp here? Can we hide it behind the feature, and assume that TB will never use our endorsements?

For the sake of endorsement, subject == attachment, it is just the same.

Reply:
I think something similar to the two implementations of verify_regex guarded by the same feature flag will work here. The non-regex implementation should through failure if an endorsement containing a regex is supplied

Reply:
LG, done.